### PR TITLE
Add update type information to custom alert delegate function

### PR DESCRIPTION
**Pull request by JussiSuojanen**
Related to issue [https://github.com/ArtSabintsev/Siren/issues/168](url)

### CODE REVIEW

File: Sources/SirenDelegate.swift [hunk 1]
```diff
@@ -8,6 +8,24 @@
...
+/// - major: Major release available: A.b.c.d
+/// - minor: Minor release available: a.B.c.d
+/// - patch: Patch release available: a.b.C.d
+/// - revision: Revision release available: a.b.c.D
+public enum UpdateType {
+    case major
+    case minor
+    case patch
+    case revision
+    case notSet
```
Comment (ArtSabintsev) [inline Sources/SirenDelegate.swift]: Rename from `notSet` to `unknown`.

File: Sources/SirenDelegate.swift [hunk 2]
```diff
@@ -28,7 +46,7 @@ public protocol SirenDelegate: class {
     func sirenDidFailVersionCheck(error: Error)
 
     /// Siren performed version check and did not display alert.
-    func sirenDidDetectNewVersionWithoutAlert(message: String)
+    func sirenDidDetectNewVersionWithoutAlert(message: String, updateType: UpdateType)
```
Comment (ArtSabintsev) [inline Sources/SirenDelegate.swift]: Deprecate the existing function using the following syntax above the existing signature, and have it point to the new signature:

```
@available(*, deprecated: 3.1.0, unavailable, renamed: "sirenDidDetectNewVersionWithoutAlert(message:, updateType:)")
func sirenDidDetectNewVersionWithoutAlert(message: String)
```

**Comment by ArtSabintsev** [comment]
Left a couple comments. You'll also want to clear the `updateType` status somewhere in the  `checkVersion` method, before doing any checking, since Siren is a Singleton.

File: Sources/SirenDelegate.swift (hunk 2 above)
Comment (JussiSuojanen) [reply Sources/SirenDelegate.swift]: I cannot mark swift protocol member as deprecated: "Siren/Sources/SirenDelegate.swift:49:6: Protocol members can only be marked unavailable in an @objc protocol."

Should I mark it deprecated in "public extension SirenDelegate"?

File: Sources/SirenDelegate.swift (hunk 1 above)
Comment (JussiSuojanen) [reply Sources/SirenDelegate.swift]: done

File: Sources/Siren.swift [hunk 3]
```diff
@@ -118,6 +121,8 @@ public final class Siren: NSObject {
             return
         }
 
+        updateType = .unknown
```
Comment (ArtSabintsev) [inline Sources/Siren.swift]: Move this above the `guard`.

File: Sources/SirenDelegate.swift (hunk 2 above)
Comment (ArtSabintsev) [reply Sources/SirenDelegate.swift]: I need to try this out myself before giving an adequate answer. Will get back to you soon.

**Comment by ArtSabintsev** [comment]
Thanks - merging this in. Will cut a release shortly.

**Comment by ArtSabintsev** [comment]
https://github.com/ArtSabintsev/Siren/releases/tag/3.1.0

It's live!

Thanks!

**Pull request merged**